### ISSUE TITLE
Complete game loop

### ISSUE THREAD

**Issue by SorraTheOrc** [TheWizardsCode/PlaygroundDecember23#38]
Most of the elements needed for a full game loop are in place, so lets get it all tied together.

- [x] Player creates a profile
- [x] Choose to enter combat
- [x] Player selects initial loadout from permanently available weapons and tools
- [x] Enter a level with 3 waves, first wave is super simple, player should emerge from it with plenty of resources and/or at least one powerup (Depends on #11)
- [x] Player completes all three waves
- [x] Player is able to buy permanent upgrades
- [x] Enter the next level, with 3 harder waves
- [x] Repeat until death
- [x] All powerups are cleared, leaving only the between level upgrades (Chris suggests `PlaygroundDecember23GameMode.DelayedDeathAction()`)
- [x] Start a new run

# Permanent upgrades

- [x] Persist the data between runs
- [x] Rebuild the loadout on hub scene
- [x] Ensure that the player has appropriate recipe ammo
- [x] Only present items that the player can afford
- [x] Only present items that the player does not have
- [x] Improve the UI
- [x] Handle the situation where there are not enough available upgrades 
- [x] Remove all weapons from the starting loadout
- [x] Give the player 150 credits on the first level so they can buy a preferred upgrade

**Comment by bsparks** [comment]
buying perm upgrades only happens after a level is complete? what do they buy perm upgrades with? if they can only buy perm upgrades after completing a level and if they require resources from the level, then they will have to survive longer and save up to buy more expensive upgrades?

**Comment by SorraTheOrc** [comment]
You get a choice of three upgrades and select one.

The in-game powerups are what cost resources.

We could add additional rewards that can be purchased with resources later, but that's out of scope for this.

**Comment by YondernautsGames** [comment]
Your above description is a bit different from how I've set up the hub scene atm. We might need a new scene for the leaderboard / reset state. I was treating the persistent data as upgrades you always have access to since that's what I'm used to in smaller scale rogue-lites (where a run is one scene rather than a sequence of scenes) - we probably need another set of upgrades that persist between scenes, but aren't saved with the player profile like this one too

**Comment by SorraTheOrc** [comment]
I'm going to remove the leaderboard from this item. It's not really part of the gameloop.

I've added the run persistent code to your existing RogueLiteManager in #40
